isis/src/base/objs/CSMCamera/CSMCamera.cpp

```diff
@@ -43,16 +43,36 @@ find files of those names at the top level of this repository. **/
 using namespace std;
 
 namespace Isis {
+
+  /**
+   * Constructor for the USGS CSM Camera Model inside ISIS.
+   */
+  CSMCamera::CSMCamera(Cube &cube, StringBlob &state) : Camera(cube) {
```

Comment:
I do not like the idea of constructing from a BLOB. BLOBs have to do with file I/O and not in memory passing data around. I would rather this construct from the actual data: plugin name, model name, and state string.

Reply:
Agreed, blob is not best here. We'll take another look at this constructor.